njs/src/tradebn/position-check.js

```diff
@@ -0,0 +1,308 @@
+
+      if (!openTpOrders.length && currentPNL > 0.25) {
+        if (side === "BUY" && Number(latestSlOrder.stopPrice) < newStopLoss) {
+          const orderIdList = openSlOrders.map(
+            (openSlOrder) => openSlOrder.orderId
+          );
+
+          await cancelOrders({
+            symbol,
+            orderIdList,
+          });
+
+          await placeStopLossOrder({
+            symbol,
+            side,
+            stopPrice: newStopLoss,
+            quantity: positionAmt,
+          });
+
+          logger.info(
+            `Stop loss updated for symbol ${symbol} to ${newStopLoss}`
+          );
+        }
+
+        if (side === "SELL" && Number(latestSlOrder.stopPrice) > newStopLoss) {
+          const orderIdList = openSlOrders.map(
+            (openSlOrder) => openSlOrder.orderId
+          );
+
+          await cancelOrders({
+            symbol,
+            orderIdList,
+          });
+
+          await placeStopLossOrder({
+            symbol,
+            side,
+            stopPrice: newStopLoss,
+            quantity: positionAmt,
+          });
+
+          logger.info(
+            `Stop loss updated for symbol ${symbol} to ${newStopLoss}`
+          );
+        }
+      }
+    }
+  } catch (error) {
+    logger.error(
+      `Failed to handle active position for symbol ${symbol}: ${error.message}`
+    );
+  }
+};
+
+const processActivePositions = async (activePositions) => {
+  for (const position of activePositions) {
+    await handleActivePosition(position);
+    await sleep(500);
+  }
+};
+
+const getAndProcessActivePositions = async () => {
+  try {
+    const positions = await client.getPositions();
+    const activePositions = positions
+      .filter(({ positionAmt }) => Number(positionAmt) !== 0)
+      .map((position) => ({
+        ...position,
+        positionAmt: Math.abs(Number(position.positionAmt)),
+        entryPrice: Number(position.entryPrice),
+        breakEvenPrice: Number(position.breakEvenPrice),
+        markPrice: Number(position.markPrice),
+        unRealizedProfit: Number(position.unRealizedProfit),
+        liquidationPrice: Number(position.liquidationPrice),
+        leverage: Number(position.leverage),
+        notional: Number(position.notional),
+        side: Number(position.positionAmt) > 0 ? "BUY" : "SELL",
+      }));
+
+    await processActivePositions(activePositions);
+  } catch (error) {
+    logger.error(
+      `Failed to retrieve and process active positions: ${error.message}`
+    );
+    throw error;
+  }
+};
+
+const flow = async () => {
+  try {
+    await getAndProcessActivePositions();
+  } catch (error) {
+    logger.error(`Flow execution failed: ${error.message}`);
+    throw error;
+  }
+};
+
+const engine = async () => {
+  const interval = 5000;
+
+  try {
+    await flow();
+  } catch (error) {
+    logger.error(`Engine execution failed: ${error.message}`);
+  }
+
+  setTimeout(engine, interval);
+};
```

Comment:
nit!
not a big thing, but maybe use sync js here instead

```javascript
const engine = () => {
  const interval = 5000;
  flow()
    .then(() => {
       setTimeout(engine, interval);
     })
     .((error) => {
        logger.error(`Engine execution failed: ${error.message}`);
        setTimeout(engine, interval);
    });
};
``` 

as the engine function is not called as promise in the index.js: 

https://github.com/cqtrade/etrade/blob/84448bb6743c05ab08302d6d79c83e1fbb66c7ff/njs/src/index.js#L12

or maybe it would be better to call engine as promise in the index.js?

Reply:
Good point! Changed it to synchronous for now, and we can later see, whether we should or need to change it to be asynchronous